src/core/vector/qgsvectorlayer.cpp

```diff
@@ -3486,7 +3467,13 @@ bool QgsVectorLayer::commitChanges( bool stopEditing )
     return false;
 
   mCommitChangesActive = true;
-  bool success = mEditBuffer->commitChanges( mCommitErrors );
+
+  bool success = false;
+  if ( mEditBuffer->editBufferGroup() )
```

Comment:
Would it make sense to move this block into `QgsEditBuffer::commitChanges()`? That would reduce the requirement to write it whenever calling `editBuffer.commitChanges()` (and potentially forgetting it).

Reply:
Mmm then i'd had to pass also `stopEditing` to `QgsEditBuffer::commitChanges()` which doesn't have any other usage for it than pass it to `QgsEditBufferGroup`. The logic of stop editing is matter of the layer.

Reply:
What happens currently if `editBuffer.commitChanges()` is called when the mode is buffered?

Reply:
The changes of that buffer will be committed. But does it make sense to call that function from other places than `QgsVectorLayer::commitChanges`?
I feel the decision (commit over group/single buffer) should stay in `QgsVectorLayer`. Then `QgsEditBuffer` does not even need to know that he his part of a group (`mEditBufferGroup` could be a direct member of `QgsVectorLayer`)